### PR TITLE
Bugfix artifacts after deletion

### PR DESCRIPTION
**Pull request by ChristianAtDell**
# Description
Previous IsiPath logic made it such that RGs would not properly delete when the IsiPath did not match between Storage Class and array Secret.  This created artifacts that made cleaning up after creating an RG difficult. Additional logic has also been added to prevent DL when deleting a Replication Group that may have existing PVs or user-created data inside of its replication folder. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| [523](https://github.com/dell/csm/issues/523) |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate (80.2% -> 80.7%)
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Test cases have been added to the replication.feature file for new method of volume group name acquisition. 

Otherwise, manual testing has been done using 'main' branch of csm-replication for replication controller/sidecar. Manual testing verified that Storage Class IsiPath no longer needs to match Secret IsiPath. It also verified that RG deletion will fail and prevent directory removal if there is anything left in the replication directory. 

Also did thorough IsiPath priority testing, to verify that SC > Secret > values.yaml priority all works as intended.

### CODE REVIEW

File: service/replication.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package service
 
 /*
- Copyright (c) 2019-2023 Dell Inc, or its subsidiaries.
+ Copyright (c) 2019-2022 Dell Inc, or its subsidiaries.
```
Comment (santhoshatdell) [inline service/replication.go:4]: 2023

File: service/replication.go [hunk 2]
```diff
@@ -730,7 +730,7 @@ func failbackDiscardLocal(ctx context.Context, localIsiConfig *IsilonClusterConf
 	log.Info("Running sync job on TGT mirror policy")
 	err = remoteIsiConfig.isiSvc.client.SyncPolicy(ctx, ppNameMirror)
 	if err != nil {
-		return status.Errorf(codes.Internal, "failback (discard local): policy sync failed %s", err.Error())
```
Comment (santhoshatdell) [inline service/replication.go:733]: Revert this change please.

File: service/replication.go [hunk 3]
```diff
@@ -826,7 +826,7 @@ func synchronize(ctx context.Context, localIsiConfig *IsilonClusterConfig, remot
 	ppName := strings.ReplaceAll(vgName, ".", "-")
 	err := localIsiConfig.isiSvc.client.SyncPolicy(ctx, ppName)
 	if err != nil {
-		return status.Errorf(codes.Internal, "sync: policy sync failed %s", err.Error())
```
Comment (santhoshatdell) [inline service/replication.go:829]: Revert this change.

**Comment by ChristianAtDell** [comment]
Closing to address some comments and a deletion issue. Will retest and reopen.

**Review by santhoshatdell — APPROVED** [review]

**Review by harshaatdell — APPROVED** [review]
